### PR TITLE
Add ability to emit literal Typescript from with Go file

### PR DESCRIPTION
**Pull request by pdhammond42**
This implements a feature to allow a Go file to contain Typescript definitions that can't be expressed through `tstype` tags.

There are actually two ways of doing it implemented. I think both have their uses but perhaps only one should be included, in which case I would say the var string literal one should be preferred.

**Motivating Example**:
We have a struct that is marshalled into JSON, and therefore exposed to TS, as a tuple of values, to minimise size on the wire. (it saves about 30% after gzip compression)

**Prior art:**
There are of course already ways to handle this. 
- we could use `frontmatter`  in the configuration, but I think this separates the TS declaration from the Go too much. 
- we could simply hand-craft the relevant TS and refer to it in an import in frontmatter, but that's an even bigger separation.
- We could put comments into the Go that describe the mapping, which keeps it close but relies on the client reading and then using them.

I hope you find this change acceptable and thanks for reading.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,6 +197,60 @@ export interface Book {
 
 You could use the `frontmatter` field in the config to inject `export type Genre = "novel" | "crime" | "fantasy"` at the top of the file, and use `tstype:"Genre"`. I personally prefer that as we may use the `Genre` type more than once.
 
+**emit directive**
```
Comment (gzuidhof) [inline README.md:200]:
```suggestion
**`tygo:emit` directive**
```

File: tygo/package_generator.go [hunk 2]
```diff
@@ -26,15 +26,25 @@ func (g *PackageGenerator) Generate() (string, error) {
 
 			// GenDecl can be an import, type, var, or const expression
 			case *ast.GenDecl:
-				if x.Tok == token.VAR || x.Tok == token.IMPORT {
+				if x.Tok == token.IMPORT {
 					return false
 				}
+				isEmit := false
+				if x.Tok == token.VAR {
```
Comment (gzuidhof) [inline tygo/package_generator.go:33]: Perhaps this makes sense to support `token.CONST` too? (or perhaps there is a good reason not to?)

File: tygo/write_toplevel.go [hunk 3]
```diff
@@ -17,6 +17,34 @@ type groupContext struct {
 	iotaValue            int
 }
 
+// isEMitVar returns true if dec is a string var with a tygo:emit directive.
```
Comment (gzuidhof) [inline tygo/write_toplevel.go:20]:
```suggestion
// isEmitVar returns true if dec is a string var with a tygo:emit directive.
```

**Review by gzuidhof — APPROVED** [review]
Looks great, thank you!

This is a great contribution :)

**Comment by gzuidhof** [comment]
Let me know when you are happy for me to merge this

File: tygo/package_generator.go (hunk 2 above)
Comment (pdhammond42) [reply tygo/package_generator.go:33]: Consts are already handled to export the const definition itself, so I'm inclined to not handle them here as it makes it more complex it we have to decide which way to handle it.

**Pull request merged**